### DISCUSSION TITLE
CPUExecutionProvider GetCapability function

### DISCUSSION THREAD

**Discussion by JAEWOOKe** [microsoft/onnxruntime#6555] [Other Q&A]
Dear all,

Hi, I have a question regarding _GetCapability_ for execution providers.

Seems like CPUExecutionProvider doesn't use the _GetCapability_ function, and can anyone tell me how nodes for CPU execution are set?

I'm looking upon the function _PartitionOnnxFormatModelImpl_ in _graph_partition.cc_.

Thanks,
Jaewook

**Comment by pranavsharma** [comment]
The CPU execution provider (EP) does implement GetCapability. The implementation sits in the base class. All ops are implemented by the CPU EP since it's meant to be a backup for other EPs.

**Reply by JAEWOOKe** [reply]
Thanks for replying. It was a dumb question. I was confused because of cases when capabilities for CPU doesn't exist but found out that KernelRegistry::TryFindKernel seems to be responsible for it.
